apps/server/src/shared/repo/task/task.repo.integration.spec.ts

```diff
@@ -985,7 +985,45 @@ describe('TaskRepo', () => {
 			});
 		});
 
-		describe('when user is the creator', () => {
+		describe('when parent is "not" passed', () => {
+			it('should "not" find finished tasks', async () => {
```

Comment:
what happens in case the task was created for a course, one lesson is attached to the task and then the course is deleted. is provided, that the lessonId is also removed correclty? Did not see any test for that but also not sure if a test is needed for this case.

Reply:
it would behave the same way as without lesson: so far it would have thrown an error, now it doesnt because the task wont be loaded in the first place.
Added a test for it.